tests/tl/test_differential_expression_analysis.py

```diff
@@ -0,0 +1,412 @@
+    with pytest.raises(KeyError):
+        dea.perform(**{DeaParameters.GROUP1: ["sample1", "sample2"]})
+
+
+def test_dea_two_groups_both_grouping_methods():
+    input_data = TestableDifferentialExpressionAnalysisTwoGroups.valid_data_input
+    dea = TestableDifferentialExpressionAnalysisTwoGroups(input_data)
+    with pytest.raises(
+        ValueError,
+        match="Please provide either a list of columns OR the grouping column, not both.",
+    ):
+        dea.perform(
+            **{
+                DeaParameters.GROUPING_COLUMN: "grouping_column",
+                DeaParameters.GROUP1: ["sample1", "sample2"],
+                DeaParameters.GROUP2: ["sample3", "sample4"],
+            }
+        )
+
+
+def test_dea_two_groups_missing_grouping_column():
+    input_data = TestableDifferentialExpressionAnalysisTwoGroups.valid_data_input
+    dea = TestableDifferentialExpressionAnalysisTwoGroups(input_data)
+    with pytest.raises(
+        ValueError, match=f"Parameter {DeaParameters.GROUPING_COLUMN} is missing."
+    ):
+        dea.perform(
+            **{
+                DeaParameters.GROUP1: "group1",
+                DeaParameters.GROUP2: "group2",
+                DeaParameters.METADATA: pd.DataFrame([[1, 2]]),
+            }
+        )
+
+
+def test_dea_two_groups_missing_metadata():
+    input_data = TestableDifferentialExpressionAnalysisTwoGroups.valid_data_input
+    dea = TestableDifferentialExpressionAnalysisTwoGroups(input_data)
+    with pytest.raises(
+        ValueError, match=f"Parameter {DeaParameters.METADATA} is missing."
+    ):
+        dea.perform(
+            **{
+                DeaParameters.GROUPING_COLUMN: "grouping_column",
+                DeaParameters.GROUP1: "group1",
+                DeaParameters.GROUP2: "group2",
+            }
+        )
+
+
+def test_dea_two_groups_get_group_members_metadata():
+    group1, group2 = DifferentialExpressionAnalysisTwoGroups._get_group_members(
+        TestableDifferentialExpressionAnalysisTwoGroups.valid_parameter_input
+    )
+    assert group1 == ["sample1", "sample2"]
+    assert group2 == ["sample3", "sample4"]
+
+
+def test_dea_two_groups_get_group_members_direct_grouping():
+    group1, group2 = DifferentialExpressionAnalysisTwoGroups._get_group_members(
+        {"group1": ["sample1", "sample2"], "group2": ["sample3", "sample4"]}
+    )
+    assert group1 == ["sample1", "sample2"]
+    assert group2 == ["sample3", "sample4"]
+
+
+def test_dea_ttest_perform_runs():
+    input_data = TestableDifferentialExpressionAnalysisTwoGroups.valid_data_input
+    dea = DifferentialExpressionAnalysisTTest(input_data)
+    dea.perform(
+        **TestableDifferentialExpressionAnalysisTwoGroups.valid_parameter_input,
+        **{
+            DeaParameters.TEST_FUN: "independent",
+            DeaParameters.FDR_METHOD: "fdr_bh",
+            PreprocessingStateKeys.LOG2_TRANSFORMED: True,
+        },
+    )
+    assert dea.result.shape == (3, 3)
```

Comment:
At least for one set of parameters, I would test the actual results.
`pd.testing.assert_frame_equals` could come handy here